acp/acp_local_test.go

```diff
@@ -128,6 +128,91 @@ func Test_LocalACP_InMemory_AddPolicy_CreatingSamePolicyAfterWipeReturnsSameID(t
 	require.Nil(t, errClose)
 }
 
+func Test_LocalACP_InMemory_AddPolicy_CreatingSamePolicyAfterDropAllReturnsSameID(t *testing.T) {
+	ctx := context.Background()
+	localACP := NewLocalACP()
+
+	localACP.Init(ctx, "")
+	errStart := localACP.Start(ctx)
+	require.Nil(t, errStart)
+
+	policyID, errAddPolicy := localACP.AddPolicy(
+		ctx,
+		identity1,
+		validPolicy,
+	)
+	require.Nil(t, errAddPolicy)
+
+	require.Equal(
+		t,
+		validPolicyID,
+		policyID,
+	)
+
+	errDrop := localACP.DropAll(ctx)
+	require.Nil(t, errDrop)
+
+	// Since nothing is persisted should allow adding same policy again with same ID
+
+	policyID, errAddPolicy = localACP.AddPolicy(
+		ctx,
+		identity1,
+		validPolicy,
+	)
+	require.Nil(t, errAddPolicy)
+	require.Equal(
+		t,
+		validPolicyID,
+		policyID,
+	)
+
+	errClose := localACP.Close()
+	require.Nil(t, errClose)
+}
+
+func Test_LocalACP_Persistent_AddPolicy_CreatingSamePolicyAfterDropAllReturnsSameID(t *testing.T) {
+	ctx := context.Background()
+	localACP := NewLocalACP()
+
+	acpPath := t.TempDir()
+	localACP.Init(ctx, acpPath)
+	errStart := localACP.Start(ctx)
+	require.Nil(t, errStart)
+
+	policyID, errAddPolicy := localACP.AddPolicy(
+		ctx,
+		identity1,
+		validPolicy,
+	)
+	require.Nil(t, errAddPolicy)
+
+	require.Equal(
+		t,
+		validPolicyID,
+		policyID,
+	)
+
+	errDrop := localACP.DropAll(ctx)
```

Comment:
nitpick: Might be interesting to also test if calling this when acp is already closed (not started) is safe or errors/panics.

Reply:
Thinking this through, should we only allow dropping if it isn't closed yet?

Reply:
Since we will soon have option to turn acp off (which can be done only by authorized admin), as long as we also plan to (which I believe we do) to gate the purge command also with authorized admin (AAC will use it's own separate local acp logic not mixed with this document access control logic), so it makes sense that the purge resets even if it's closed.

However happy to discuss further. We could also have a --force flag with purge that does the behavior of purging on close and otherwise a "safer" once that will error if acp is off (or is closed for whatever reason).

Reply:
Let's not forget that the purge command is a dev only feature so protecting it with Admin ACP doesn't really make sense.

Reply:
> Let's not forget that the purge command is a dev only feature so protecting it with Admin ACP doesn't really make sense.

Kind of forgot about that. Do we still want dev only features when aac is implemented?

Reply:
I think so yes. I would never want purge to be accessible by anyone on a production DB.